pkg/validate/astrolabevalidator.go

```diff
@@ -0,0 +1,76 @@
+package validate
+
+import (
+	"fmt"
+	"log"
+)
+
+type AstrolabeCompatibilityValidator struct{}
+
+func (v *AstrolabeCompatibilityValidator) IsPluginCompatibleWithPlatform(platformVersion string, pluginId string, pluginVersion string, repos []PluginRepository) (Verdict, error) {
+	var metadata *CompatibilityMetadata
+	for _, e := range repos {
+		m, err := e.getCompatibilityMetadata(pluginId)
+		if err != nil {
+			log.Printf("error when geting compatibility metadata: %s", err)
+		} else {
+			metadata = m
+			break
+		}
+	}
+
+	if metadata == nil {
+		return Unknown, fmt.Errorf("compatibility metadata not found in repositories")
+	}
+
+	tests, err := metadata.getCompatibilityTests(pluginVersion)
+	if err != nil {
+		return Unknown, err
+	}
+
+	for _, e := range tests {
+		if e.containsPlatformVersion(platformVersion) {
```

Comment:
What about a plugin that operates in two (or more) services? This will eventually need to check that too.

Reply:
this is missing in my implementation but I created a PR to fix it.